### PR TITLE
Delete Contribution Flow(frontend)

### PR DESCRIPTION
**Pull request by NKoech123**
## Linear Ticket
PRO-287: https://linear.app/govrn/issue/PRO-287/delete-icon-not-deleting-staged-contributions

## Description

 @jonathanprozzi  
Updates: 
  - I added a component in **apps/protocol-frontend/src/components/DeleteContributionDialog.tsx**  that  triggers
     a dialog when the delete icon is clicked. The dialog enables a User to either cancel the deleting process or proceed.
     Please refer to the screenshots below.
Screen Shot 2022-08-03 at 5 51 32 PM
Later changed the "Delete" Button to "Confirm".  I thought "Confirm" makes more sense.
Screen Shot 2022-08-03 at 9 49 39 PM

Contribution Successfully deleted!
Screen Shot 2022-08-03 at 12 52 04 AM

### CODE REVIEW

**Comment by jonathanprozzi** [comment]
Thanks @NKoech123 ! I'll take a look. I think I know what's going on with this, we need to do a background refresh of the Contributions after the delete -- going to see, and then I can show you what's up if this is what's going on with it.

**Comment by NKoech123** [comment]
Thanks @jonathanprozzi !!

File: apps/protocol-frontend/src/components/ContributionsTable.tsx [hunk 1]
```diff
@@ -158,16 +161,13 @@ const ContributionsTable = ({
...
-                  icon={<FiTrash2 fontSize="1rem" />}
-                  variant="ghost"
-                  color="gray.800"
-                  disabled={
-                    row.original.user.id !== userData?.id ||
-                    row.original.status === 'minted'
-                  }
-                  aria-label="Delete Contribution"
-                />
+                <DeleteContributionDialog
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:164]: Hmm, not sure why we'd want to split this off from the Table itself
The delete icon button should be able to call the Dialog

File: apps/protocol-frontend/src/components/DeleteContributionDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        
+        />
+        <AlertDialog
+          isOpen={isOpen}
+          leastDestructiveRef={cancelRef}
+          onClose={onClose}
+          size = {DialogSize}
+        >
+          <AlertDialogOverlay >
+            <AlertDialogContent bgColor="brand.primary.50" color="gray.800">
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/DeleteContributionDialog.tsx:61]: lets use `bgColor='white'` like our other modals in the UI

File: apps/protocol-frontend/src/components/DeleteContributionDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                <Button 
+                 ref={cancelRef} 
+                 onClick={onClose}>
+                  Cancel
+                </Button>
+
+                <Button 
+                 colorScheme='red' 
+                 color="brand.primary.600"
+                 backgroundColor="brand.primary.100"
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/DeleteContributionDialog.tsx:77]: I think we may want to use red for this -- remove the `backgroundColor`

**Comment by jonathanprozzi** [comment]
@NKoech123 A few comments about the styles (we should stick with white for the dialog since it's a similar UI pattern to our modals -- appreciate that you tried 3 versions!)

The state issues are from the button + handlers being moved into the Dialog itself -- let's keep that split and pass the handler from the ContributionTable -> AlertDialog via the delete icon. So we'll want to add that delete icon back to the Table.

File: apps/protocol-frontend/src/components/ContributionsTable.tsx (hunk 1 above)
Comment (NKoech123) [reply apps/protocol-frontend/src/components/ContributionsTable.tsx]: Thanks for the feedback. I made some changes based on your comments above (letting the icon button call the Dialog)

File: apps/protocol-frontend/src/components/DeleteContributionDialog.tsx (hunk 2 above)
Comment (NKoech123) [reply apps/protocol-frontend/src/components/DeleteContributionDialog.tsx:40]: Addressed bgColor

**Comment by NKoech123** [comment]
Subject: Issue with disabled Action icons Solved.
Hi @jonathanprozzi , I had an issue where I would report a contribution, and the edit/delete action icons would stay disabled until the page is manually refreshed. I did some debugging and saw that `row.original.user.id` is undefined but  `row.original.user` gives you the user id. 

So I changed the code in the IconButton from
  ` disabled={
                  row.original.user.id!== userData?.id
                }
`
to 
  ` disabled={
                  row.original.user!== userData?.id
                }
`

**Comment by jonathanprozzi** [comment]
I'm running this locally and I'm not able to edit/delete any of my contributions -- do you have this locally @NKoech123

**Comment by NKoech123** [comment]
I was able to do all the editing/deleting yesterday and all worked well. Let me jumped on my desk now. Will get back to you in a few minutes

File: apps/protocol-frontend/src/components/DeleteContributionDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                <Button 
+                 ref={cancelRef} 
+                 onClick={() => setDialog({ ...dialog, isOpen: false })} 
+                 >
+                  Cancel
+                </Button>
+
+                <Button 
+                 colorScheme='red' 
+                 color="brand.primary.600"
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/DeleteContributionDialog.tsx:64]: we need to remove the `color` and the `backgroundColor` -- they're overriding the `colorScheme`

**Comment by jonathanprozzi** [comment]
@NKoech123 some of the Prettier settings seem different too -- I'm seeing a lot of extra lines and spaces in the PR. Is everything formatting on save?

File: apps/protocol-frontend/src/components/ContributionsTable.tsx [hunk 5]
```diff
@@ -56,6 +62,31 @@ const ContributionsTable = ({
...
+  useEffect(() => {
+            setDialog(dialog)
+               }, [dialog])
+
+  
+  const handleDeleteContribution = (contribution_id: string) =>{
+    setDialog({
+      ...dialog,
+      isOpen: true,  //this opens AlertDialog
+      title: "Are you sure you want to delete this contribution? You can't undo this action afterwards",
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/src/components/ContributionsTable.tsx:83]: Let's change the title to: `"Are you sure you want to delete this Contribution? You can't undo this action."

We tend to capitalize Contribution thoughout the app

**Review by jonathanprozzi — COMMENTED** [review]
Nice work overall! Delete works great. Have some questions I left as comments throughout.

**Comment by NKoech123** [comment]
I pull from the staging to my branch and I am experience error below:
I dont know if it's something you have seen
Screen Shot 2022-08-10 at 10 23 03 AM

File: apps/protocol-frontend/src/components/DeleteContributionDialog.tsx (hunk 4 above)
Comment (NKoech123) [reply apps/protocol-frontend/src/components/DeleteContributionDialog.tsx:64]: I just removed those. Thanks!

File: apps/protocol-frontend/src/components/ContributionsTable.tsx (hunk 5 above)
Comment (NKoech123) [reply apps/protocol-frontend/src/components/ContributionsTable.tsx:83]: Comment addressed. I have updated the title.

**Comment by NKoech123** [comment]
@jonathanprozzi Thanks for pointing out the extra spaces in the PR. I just noticed that's not a good practice. That's due to me altering pre-existing code and not checking line by line before pushing. I have taken note of that.

**Comment by NKoech123** [comment]
@jonathanprozzi  I also saw your comment above about not being able to edit/delete contribution on your side. Initially I didn't have this issue. Now I am seeing this issue. Let me debug the issue at the moment

**Comment by NKoech123** [comment]
_> @jonathanprozzi I also saw your comment above about not being able to edit/delete contribution on your side. Initially I didn't have this issue. Now I am seeing this issue. Let me debug the issue at the moment_

And finally the actions icons are back and functioning. Let me know if you encounter something interesting on your side.

Screen Shot 2022-08-10 at 3 27 05 PM

**Comment by jonathanprozzi** [comment]
Still seeing an odd bug with the Edit and Delete -- it's fine on refresh/initial load and then disabled after navigating back and forth:


https://user-images.githubusercontent.com/9438776/184261600-cbea3d2c-698a-4545-a2f7-36f138825ccb.mp4

**Comment by NKoech123** [comment]
I see. Let me take a look at it.

**Comment by jonathanprozzi** [comment]
@alexkeating @NKoech123 - resolved merge conflicts and fixed a few small things. Will fix the type issue too before this is ready

Also, Keating -- the fix you put up earlier with the commit I added on top resolves the remaining bug with this, so we can merge yours first and then this one

**Pull request merged**